packages/test/src/test-ai-sdk.ts

```diff
@@ -0,0 +1,322 @@
+/**
+ * Test AI SDK integration with Temporal workflows
+ */
+import { LanguageModelV2Content, LanguageModelV2FinishReason } from '@ai-sdk/provider';
+import { openai } from '@ai-sdk/openai';
+import { v4 as uuid4 } from 'uuid';
+import * as opentelemetry from '@opentelemetry/sdk-node';
+import { SEMRESATTRS_SERVICE_NAME } from '@opentelemetry/semantic-conventions';
+import { ExportResultCode } from '@opentelemetry/core';
+import { StdioClientTransport } from '@modelcontextprotocol/sdk/client/stdio.js';
+import { experimental_createMCPClient as createMCPClient } from '@ai-sdk/mcp';
+import { AiSDKPlugin, ModelResponse, TestProvider } from '@temporalio/ai-sdk';
+import { temporal } from '@temporalio/proto';
+import { WorkflowClient } from '@temporalio/client';
+import {
+  makeWorkflowExporter,
+  OpenTelemetryActivityInboundInterceptor,
+  OpenTelemetryActivityOutboundInterceptor,
+  OpenTelemetrySinks,
+  OpenTelemetryWorkflowClientCallsInterceptor,
+  OpenTelemetryWorkflowClientInterceptor,
+} from '@temporalio/interceptors-opentelemetry';
+import { InjectedSinks, Runtime } from '@temporalio/worker';
+import {
+  generateObjectWorkflow,
+  helloWorldAgent,
+  mcpWorkflow,
+  middlewareWorkflow,
+  telemetryWorkflow,
+  toolsWorkflow,
+} from './workflows/ai-sdk';
+import { helpers, makeTestFunction } from './helpers-integration';
+import { getWeather } from './activities/ai-sdk';
+import { Worker } from './helpers';
+import EventType = temporal.api.enums.v1.EventType;
+
+const remoteTests = !!process.env.AI_SDK_REMOTE_TESTS;
+
+function contentResponse(
+  content: LanguageModelV2Content[],
+  finishReason: LanguageModelV2FinishReason = 'stop'
+): ModelResponse {
+  return {
+    content,
+    finishReason,
+    usage: {
+      inputTokens: undefined,
+      outputTokens: undefined,
+      totalTokens: undefined,
+    },
+    warnings: [],
+  };
+}
+
+function textResponse(content: string): ModelResponse {
+  return contentResponse([
+    {
+      type: 'text',
+      text: content,
+    },
+  ]);
+}
+
+function toolCallResponse(toolName: string, input: string): ModelResponse {
+  return contentResponse(
+    [
+      {
+        type: 'tool-call',
+        toolCallId: 'call_yY3nlDwH5BQSJo63qC61L4ZB',
+        toolName,
+        input,
+      },
+    ],
+    'tool-calls'
+  );
+}
+
+function* helloWorkflowGenerator(): Generator<ModelResponse> {
+  yield textResponse('Test Haiku');
+}
+
+const test = makeTestFunction({
+  workflowsPath: require.resolve('./workflows/ai-sdk'),
+});
+
+test('Hello world agent responds in haikus', async (t) => {
+  t.timeout(120 * 1000);
```

Comment:
I'm a bit worried about setting individual test timeout to 2 minutes. That means that eventual bugs that would prevent a worker from shutting down will cause the CI job to time out while running this test (which runs very early because `ai` starts with an `a`), with almost no context on what failed.

I suppose that's only pertinent when `remoteTests` is true, right?
And `remoteTests` would be false in CI, right?
If so, can we only bump up the test timeout to 2 minutes when remote tests are enabled?